bin/brew

```diff
@@ -45,5 +85,77 @@ fi
 
 export HOMEBREW_BREW_FILE
 export HOMEBREW_RUBY_PATH
+export HOMEBREW_PREFIX
+export HOMEBREW_REPOSITORY
+export HOMEBREW_LIBRARY
+export HOMEBREW_CELLAR
+
+if [ -n "$HOMEBREW_OSX" ]
+then
+  if [ "$(xcode-\select --print-path)" = "/" ]
+  then
+    cat >&2 <<EOS
```

Comment:
> I think your `odie` there is significantly harder to follow […]

Do you mean this piece of code `IFS= read -rd '' message`? Because everything else in there is trivial. I think it's not overly complicated and I think it's a good thing to use `odie` everywhere for consistency (and one won't have to remember to always add the `>&2`). As Xu says, this also makes it easier to add highlighting (which can be added later).